tools/closed-by-commits

```diff
@@ -0,0 +1,436 @@
+#!/usr/bin/env -S uv run --script --frozen --only-group ci  # -*-python-*-
+
+import json
+import re
+import time
+import urllib.parse
+from collections import defaultdict
+from dataclasses import dataclass, field
+from functools import cache
+from typing import Annotated, Any
+
+import requests
+import typer
+from github import Auth, Github
+from rich.console import Console
+from rich.progress import Progress
+from typing_extensions import override
+
+
+def encode_hash_component(s: str) -> str:
+    hash_replacements = {
+        "%": ".",
+        "(": ".28",
+        ")": ".29",
+        ".": ".2E",
+    }
+    encoded = urllib.parse.quote(s, safe="*")
+    return "".join(hash_replacements.get(c, c) for c in encoded)
+
+
+@cache
+def search_czo_for_number(prefix: str, number: int) -> frozenset[str]:
+    params = {
+        "anchor": "newest",
+        "num_before": "100",
+        "num_after": "0",
+        "narrow": json.dumps(
+            [
+                {"negated": False, "operator": "search", "operand": f'"#{prefix}{number}"'},
+                {"negated": False, "operator": "channels", "operand": "web-public"},
+            ]
+        ),
+    }
+
+    try:
+        response = requests.get("https://chat.zulip.org/json/messages", params=params, timeout=30)
+        response.raise_for_status()
```

Comment:
Probably a comment should say why we're using this rather than the Python bindings, so someone thinking about reworking this to use that can know what problem(s) need to be solved to do that?

Reply:
I don't particularly think that the Python bindings add much here.  I don't have any particular reason to not use them; I'm sure they'd be fine.